### PR TITLE
Allow missing window/workDoneProgress/create request from the server

### PR DESCRIPTION
**Pull request by rwols**
This commit adds a workaround for servers that report server-initiated progress but don't request window/workDoneProgress/create before reporting progress.

I haven't actually tested whether this works with jdlts regarding its progress reporting behavior in https://github.com/sublimelsp/LSP-jdtls/issues/37, could you check @LDAP ?

### CODE REVIEW

**Comment by rchl** [comment]
It's nothing new in this PR but we shouldn't be relying on destructor for updating UI as destructor is not guaranteed to run immediately and that can result in progress lingering for longer than necessary. For example this was easily visible before (and still is for some features) when restarting the server. So regions would keep showing for a second or so after restarting a server on my system.

**Comment by rwols** [comment]
I understand the concern but I don't quite know how to address it.

File: plugin/active_request.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from .core.sessions import SessionViewProtocol
```
Comment (rchl) [inline plugin/active_request.py:1]: Shouldn't this file live in `core/` since it only deals with core types?

**Review by LDAP — APPROVED** [review]
PR fixes the issues with JDTLS.

File: plugin/active_request.py (hunk 1 above)
Comment (rchl) [reply plugin/active_request.py:1]: @rwols you want me to address that?

Comment (rwols) [reply plugin/active_request.py:1]: No let me do it :D I'll get to it this evening, I hope.

**Review by rchl — APPROVED** [review]

**Pull request merged**